qt/Session.cc

```diff
@@ -1060,23 +1060,23 @@ void Session::addTorrent(AddData add_me, tr_variant* args_dict, bool trash_origi
                 }
             }
 
-            if (session_has_torrent && !add_me.filename.isEmpty())
+            if (auto const& filename = add_me.filename;
+                session_has_torrent && !filename.isEmpty() && add_me.type == AddData::FILENAME)
             {
-                QFile(add_me.filename).rename(QStringLiteral("%1.added").arg(add_me.filename));
+                auto file = QFile{ filename };
+
+                if (trash_original)
+                {
+                    file.setPermissions(QFile::ReadOwner | QFile::WriteOwner);
+                    file.remove();
+                }
+                else
+                {
+                    file.rename(QStringLiteral("%1.added").arg(filename));
```

Comment:
Question: why does it rename the file at all? This is not something I, as an user, expect or want. It looks like this can't be disabled either.